hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -135,10 +147,69 @@ public void resume() {
   @Override
   public BackgroundTaskQueue getTasks() {
     BackgroundTaskQueue queue = new BackgroundTaskQueue();
-    queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    deletedDirSupplier.reInitItr();
```

Comment:
can avoid reinit iterator here, can be done on need bases, like initially null. In get(), if null or or hasNext is false, can init there.